### PR TITLE
Decouple the redis useage in the lock from Resque

### LINKED ISSUE #1882: replace job locking via resque for switch to sidekiq
From John in slack on May 11, 2022

>One other thing we'd need to do for sidekiq is port pres cat's job locking, which uses resque's redis methods.  We could either switch to using the redis gem directly (since that's also an indirect dependency of sidekiq anyway).  Or we could try a gem like https://github.com/veeqo/activejob-uniqueness (which at a glance seems maintained, but is just one recommendation from a random blog post found when searching "queue deduplication sidekiq", and the current job locking code has a pretty small footprint anyway).

image

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔

Fixes #1882

## How was this change tested? 🤨

Ran https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_preassembly_image_spec.rb

### CODE REVIEW

**Comment by ndushay** [comment]
I'm requesting that @jmartin-sul review this.

File: config/resque.yml [hunk 1]
```diff
@@ -1 +1,2 @@
 development: <%= ENV.fetch('REDIS_URL', 'localhost:6379') %>
+test: <%= ENV.fetch('REDIS_URL', 'localhost:6379') %>
```
Comment (ndushay) [inline config/resque.yml:2]: would we need this for prod as well?

**Comment by ndushay** [comment]
this was causing errors on -stage

File: config/resque.yml (hunk 1 above)
Comment (jcoyne) [reply config/resque.yml:2]: no because config/resque.yml is in shared configs.

**Comment by jcoyne** [comment]
> this was causing errors on -stage

I refactored this so we're not managing the redis connection ourselves and I think things have improved.

**Review by jmartin-sul — APPROVED** [review]

**Pull request merged**